libarchive/archive_read_support_format_7zip.c

```diff
@@ -1089,7 +1089,7 @@ init_decompression(struct archive_read *a, struct _7zip *zip,
 				filters[fi].id = LZMA_FILTER_DELTA;
 				memset(&delta_opt, 0, sizeof(delta_opt));
 				delta_opt.type = LZMA_DELTA_TYPE_BYTE;
-				delta_opt.dist = 1;
+				delta_opt.dist = coder2->properties[0] + 1;
```

Comment:
(1) I'm not sure if anything validates that propertiesSize == 1 here, so that should be checked.
(2) A test case is needed.

Reply:
Done.